File: DailyChallenges/CPP/LeetCode_121.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int _lmax = 0, ans = 0;
        for (int i = prices.size()-1; i>=0; --i){
            _lmax = max(_lmax, prices[i]);
            ans = max(ans, _lmax-prices[i]);
        }
        return ans;
    }
};

File: DailyChallenges/Scala/LeetCode_121.scala
object Solution {
    def maxProfit(prices: Array[Int]): Int = {
        var lmax = 0; var ans = 0;
        for (i <- 1 until prices.length){
            lmax += prices(i)-prices(i-1)
            lmax = Math.max(0, lmax)
            ans = math.max(ans, lmax);
        }
        ans
    }
}
